src/main/java/org/spongepowered/api/event/world/ExplosionEvent.java

```diff
@@ -89,4 +90,39 @@
      * updated and/or changed.
      */
     interface Post extends ExplosionEvent, ChangeBlockEvent.Post {}
-}
+
+    /**
+     * An event that is fired when the Explosion fetches the explosion resistance of a block.
+     * Provides the world, block position, block state, and current explosion resistance.
+     */
+    interface FetchBlockExplosionResistance extends ExplosionEvent, TargetWorldEvent {
```

Comment:
Why does this need to be a separate event? Why can't such things go in `Pre` and/or `Detonate`?

Reply:
`Pre` fires before the rays are cast and block resistances checked.
`Detonate` fires after the blocks have already been processed and would mean having to do all the ray casting again.

Reply:
This even it going to get expensive because you've done it on a per block basis. I also don't really agree with the idea of exposing resistances when we could have a proper filter on `Detonate`. 

Why do you want to change the resistance? If it is just to guarantee that a block will be removed (or vice versa) then it really should be in `Detonate`. Indeed, I'd just prefer capturing all blocks that have been processed by the explosion, whether or not they would explode, and provide them in `Detonate`, with whether or not they're going to be removed. A plugin can then work though that list and decide if that needs changing, even with a random number generator if required.

If this event is to stay (someone else can decide that), this needs to be a bulk event, not a per block event. Constructing and firing this event hundereds of times in a tick (potentially) is going to _really_ hurt the performance of the server.

Reply:
You'd have to capture all blocks in the radius of the explosion, not just the ones the explosion is currently effecting. Maybe this would be more optimised.

Secondly, using `Detonate` in my use case, would require running raytracing in the listener after the source code and rerunning the entity knockback too.

Reply:
> Why do you want to change the resistance?

Please answer my question. All you've done is say "in my use case" and come up with some reason why this doesn't work for you without explaining what it is you're trying to do. The question has been asked for a reason - to get an idea of what you _really_ want to do and so figure out what, if any, a better solution is.

Reply:
Also, I think `Pre` might be getting triggered in the wrong place?

Because I would suggest putting this better way of doing it there but it is located with the WorldServerMixin but would be needed within the ExplosionMixin for this.

Reply:
> > Why do you want to change the resistance?
> 
> Please answer my question. All you've done is say "in my use case" and come up with some reason why this doesn't work for you without explaining what it is you're trying to do. The question has been asked for a reason - to get an idea of what you _really_ want to do and so figure out what, if any, a better solution is.

Sorry, I missed that part and this is where I struggle to explain. To best explain what I am trying to achieve by adding this feature to the API I should probably start by explaining my plugin.

My plugin is a port of an existing Spigot plugin that allows players to pilot and move a collection of blocks called a "craft". Some examples of this:

![](https://pbs.twimg.com/media/Df05IK1UYAAY3bw.jpg)
![](https://static.planetminecraft.com/files/resource_media/screenshot/1531/2015-07-27_1731139201469.jpg)

One of its features is to override the explosion resistance of blocks to create various armour types for these "crafts".

I want to implement this feature and expand it so that by using a config setting blocks can be made stronger or weaker, and blocks that are located within a "craft" receive a bonus to explosion resistance based on the size of the "craft".

If you want different details or a better explanation in a specific area of the above I will need help to understand what you require.

Reply:
While working to move it to a single event per explosion I have discovered that explosions continue a given raytrace outside of the explosion radius if they do not meet enough resistance.

This could be optimized so that it stops if it exits the radius, however, this may have unforeseen side-effects. What would you advise?